docs/developer-docs/smart-contracts/call/arguments.mdx

```diff
@@ -194,11 +194,86 @@ fn get_text(d: Day) -> &'static str {
 <AdornedTab value={"typescript"} label="TypeScript" endAdornment={<BetaChip />}>
 
 ```typescript
+import { IDL } from 'azle';
+
 export const ReactionType = IDL.Variant({
     Fire: IDL.Null,
     ThumbsUp: IDL.Null,
     ThumbsDown: IDL.Null
 });
+export type ReactionType =
```

Comment:
I was thinking it might be best for this example to almost exactly match the other two variant examples, I'm not sure if the variant branches should have null or string types, this one has string types:

```typescript
import { IDL, query } from 'azle';

const Day = IDL.Variant({
    Sun: IDL.Text,
    Mon: IDL.Text,
    Tue: IDL.Text,
    Wed: IDL.Text,
    Thu: IDL.Text,
    Fri: IDL.Text,
    Sat: IDL.Text
});
type Day =
    | { Sun: string }
    | { Mon: string }
    | { Tue: string }
    | { Wed: string }
    | { Thu: string }
    | { Fri: string }
    | { Sat: string };

export default class {
    @query([Day], IDL.Text)
    getText(day: Day): string {
        if ('Sun' in day) {
            return day.Sun;
        }

        if ('Mon' in day) {
            return day.Mon;
        }

        if ('Tue' in day) {
            return day.Tue;
        }

        if ('Wed' in day) {
            return day.Wed;
        }

        if ('Thu' in day) {
            return day.Thu;
        }

        if ('Fri' in day) {
            return day.Fri;
        }

        if ('Sat' in day) {
            return day.Sat;
        }

        throw new Error(`Variant "Day" has unknown tag`);
    }
}
```

Reply:
Revised example